include/core/specific_heat_model.h

```diff
@@ -0,0 +1,177 @@
+ */
+class SpecificHeatModel
+{
+public:
+  /**
+   * @brief Returns the specific heat
+   *
+   * @param temperature Temperature at time t+dt
+   *
+   * @param previous_temperature Temperature at time t
+   */
+  virtual double inline get_specific_heat(
+    const double temperature,
+    const double previous_temperature) = 0;
+};
+
+
+/**
+ * @brief Constant specific heat. Returns a constant specific
+ * heat for a fluid
+ */
+class SpecificHeatConstant : public SpecificHeatModel
+{
+public:
+  /**
+   * @brief Default constructor
+   */
+  SpecificHeatConstant(const double p_specific_heat)
+    : specific_heat(p_specific_heat)
+  {}
+
+  /**
+   * @brief Returns the specific heat
+   *
+   * @param temperature Temperature at time t+dt
+   *
+   * @param previous_temperature Temperature at time t
+   */
+  virtual double inline get_specific_heat(
+    const double /*temperature*/,
+    const double /*previous_temperature*/) override
+  {
+    return specific_heat;
+  }
+
+  const double specific_heat;
+};
+
+
+/**
+ * @brief Constant specific heat. Returns a constant specific
+ * heat for a fluid
+ */
+class PhaseChangeSpecificHeat : public SpecificHeatModel
+{
+public:
+  /**
+   * @brief Default constructor
+   */
+  PhaseChangeSpecificHeat(
+    const Parameters::PhaseChange p_phase_change_parameters)
+    : param(p_phase_change_parameters)
+  {}
+
+  /**
+   * @brief Returns the specific heat
+   *
+   * @param temperature Temperature at time t+dt
+   *
+   * @param previous_temperature Temperature at time t
+   */
+  virtual double inline get_specific_heat(
+    const double temperature,
+    const double previous_temperature) override
+  {
+    if (temperature > previous_temperature)
+      {
+        const double dT = std::max(temperature - previous_temperature, 1e-6);
+        return (enthalpy(temperature) - enthalpy(previous_temperature)) / dT;
+      }
+    else
+      {
+        const double dT = std::max(previous_temperature - temperature, 1e-6);
+        return (enthalpy(previous_temperature) - enthalpy(temperature)) / dT;
+      }
+  }
+
+  /**
+   * @brief solid_fraction Calculates the solid fraction of a phase change material at
+   * a temperature T
+   *
+   * @param T temperature at which to calculate the solid fraction
+   *
+   */
+  inline double
+  solid_fraction(const double T)
+  {
+    return std::min(std::max((T - param.T_solidus) /
+                               (param.T_liquidus - param.T_solidus),
+                             0.),
+                    1.);
+  }
+
+
+  /**
+   * @brief enthalpy Calculates the enthalpy of a phase change material for a temperature T
+   * The enthalpy is defined as :
+   *
+   * !! Pure liquid !!
+   * if (T>T_liquidus) : H = cp_solid * T_solidus + 0.5*(cp_solid+cp_liquid) *
+   * (T_liquidus-T_solidus) + latent_enthalpy + cp_liquid * (T-T_liquidus)
+   *
+   * !! Liquid-solid mix !!
+   * else if (T>T_solidus) : cp_solid * T_solidus + 0.5*(cp_solid+cp_liquid) *
```

Comment:
No because this would mean that the thermodynamic path would not be consistent. The underlying hypothesis is that the Cp is the average of both phase are you are undergoing phase change, it's just a way to smooth the transition because anyway in the temperature between liquidus and solidus, phase change dominate the Cp.